src/backend/cuda/device_manager.cpp

```diff
@@ -105,50 +126,66 @@ bool checkDeviceWithRuntime(int runtime, pair<int, int> compute) {
 }
 
 /// Check for compatible compute version based on runtime cuda toolkit version
-void checkAndSetDevMaxCompute(pair<int, int> &prop) {
-    auto originalCompute = prop;
-    UNUSED(originalCompute);
-    int rtCudaVer = 0;
+void checkAndSetDevMaxCompute(pair<int, int> &computeCapability) {
+    auto originalCompute = computeCapability;
+    int rtCudaVer        = 0;
     CUDA_CHECK(cudaRuntimeGetVersion(&rtCudaVer));
     auto tkitMaxCompute = find_if(
         begin(Toolkit2MaxCompute), end(Toolkit2MaxCompute),
         [rtCudaVer](cuNVRTCcompute v) { return rtCudaVer == v.cudaVersion; });
 
+    bool embeddedDevice = isEmbedded(computeCapability);
+
     // If runtime cuda version is found in toolkit array
     // check for max possible compute for that cuda version
     if (tkitMaxCompute != end(Toolkit2MaxCompute) &&
-        prop.first > tkitMaxCompute->major) {
-        prop = make_pair(tkitMaxCompute->major, tkitMaxCompute->minor);
-#ifndef NDEBUG
-        char errMsg[] =
-            "Current device compute version (%d.%d) exceeds supported maximum "
-            "cuda runtime compute version (%d.%d). Using %d.%d.";
-        fprintf(stderr, errMsg, originalCompute.first, originalCompute.second,
-                prop.first, prop.second, prop.first, prop.second);
-#endif
-    } else if (prop.first > Toolkit2MaxCompute[0].major) {
+        computeCapability.first >= tkitMaxCompute->major) {
+        int minorVersion = embeddedDevice ? tkitMaxCompute->embedded_minor
+                                          : tkitMaxCompute->minor;
+
+        if (computeCapability.second > minorVersion) {
```

Comment:
Does this condition actually help? I mean, if major version is greater, minor version doesn't matter I think.

Similar check in the next elseif give path

Reply:
for 9.0 the max compute capability is 7.0 but my card is 7.5. This configuration causes compilation errors when building with nvrtc like the one I posted in my description.

Reply:
Right, I didn't notice the equal sign in >=. My bad